core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -123,6 +126,48 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function. */
+  public static String toBase64(String string) {
+    String base64 = Base64.getEncoder().encodeToString(string.getBytes(UTF_8));
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    int i;
+    if (base64.length() > 76) {
+      for (i = 0; i + 76 < base64.length(); i += 76) {
+        str.append(base64, i, i + 76);
+        str.append("\n");
+      }
+      str.append(base64, i, base64.length());
+      return str.toString();
+    }
+    return base64;
+  }
+
+  /** SQL TO_BASE64(string) function for binary string. */
+  public static String toBase64(ByteString string) {
+    String base64 = Base64.getEncoder().encodeToString(string.getBytes());
+    StringBuilder str = new StringBuilder(base64.length() + base64.length() / 76);
+    int i;
+    if (base64.length() > 76) {
+      for (i = 0; i + 76 < base64.length(); i += 76) {
+        str.append(base64, i, i + 76);
+        str.append("\n");
+      }
+      str.append(base64, i, base64.length());
+      return str.toString();
+    }
+    return base64;
+  }
+
+  /** SQL FROM_BASE64(string) function. */
+  public static ByteString fromBase64(String base64) {
+    try {
+      base64 = base64.replaceAll("[\\t\\n\\r\\s]", "");
+      return new ByteString(Base64.getDecoder().decode(base64.getBytes(UTF_8)));
```

Comment:
This looks a bit strange.  Why do you use `base64.getBytes(UTF_8)` here?
Isn't `Base64.getDecoder().decode(base64)` enough?